### DISCUSSION TITLE
How can I disable searching for headers in the Visual Studio folder?

### DISCUSSION THREAD

**Discussion by luizfernandonb** [clangd/clangd#2106] [Q&A] [answered]
I have a VS Code project, and when I press F12 to go to a header definition, clangd goes to a Visual Studio header that I have installed on my PC, but I wanted to go to the Clang header itself, how can I do this?

![image](https://github.com/user-attachments/assets/e02e2f21-f201-42df-a9c5-e4a311c4915c)

![image](https://github.com/user-attachments/assets/404e11be-b105-4e1e-867b-7b3cf66521a8)

![image](https://github.com/user-attachments/assets/a776666a-48cc-4b3e-9558-bc96b3154606)

**Comment by HighCommander4** [comment]
By "clang headers", do you mean that you are building the project with libc++?

There are three widely used implementations of the C++ standard library: MSVC, libstdc++, and libc++. Clang is capable of using any of them.

**Reply by luizfernandonb** [reply]
"Clang headers" i'm talking about are the standard C libraries that are installed in the "C:\Program Files\Clang\lib\clang\18\include" folder

**Reply by luizfernandonb** [reply]
I managed to "solve" it by uninstalling the Visual Studio IDE, so clangd found the header files in the "C:\Program Files\Clang\lib\clang\18\include" folder, but I wish there was some way to use this folder by default without having to mess with Visual Studio

**Reply by HighCommander4** [reply]
I don't think there is a complete set of C standard library headers in `C:\Program Files\Clang\lib\clang\18\include`. For example, I don't think there is `stdio.h` there. (I'm basing this on looking at what I believe is the Linux equivalent, `/usr/lib/llvm-18/lib/clang/18/include/`.)

Even in the case of `stdint.h`, the file in that directory is just a wrapper for the "real" `stdint.h`, which may be the MSVC one depending on which standard library you're actually using. (And the definition of `uint64_t` is contained in the "real" `stdint.h`.)

What compiler are you using to build your code?

**Reply by luizfernandonb** [reply]
I'm using clang, and I'm compiling it freestanding, without the C library

**Reply by HighCommander4** [reply]
> I'm using clang, and I'm compiling it freestanding, without the C library

Ok, I see. I think your problem is that clangd does not know you're compiling in freestanding mode.

Presumably that involves passing a certain flag to clang, but in the screenshot of the log I see "with command clangd fallback", which suggests you haven't told clangd what flags you're building the code with.

Please have a look at https://clangd.llvm.org/installation#project-setup to see how to tell clangd about your compile flags.

**Reply by luizfernandonb** [reply]
In this case I'm using ninja, doesn't this mean the flags will work?

**Reply by HighCommander4** [reply, marked as answer]
I'm not very familiar with ninja, but based on a quick look at https://ninja-build.org/manual.html#_extra_tools, it looks like it has a `compdb` command for generating a `compile_commands.json` file.

Did you run that command? Where is the `compile_commands.json` file located in the filesystem?

**Reply by luizfernandonb** [reply]
It worked, thanks, I didn't know I had to do this step-by-step
